### PR TITLE
Make parmmg dependent on mmg and metis

### PR DESCRIPTION
**Pull request by lrtfm**
1. copy source of mmg to installation tree
2. add cmake args for parmmg
3. add '+mmg' when '+parmmg' for petsc

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot maintainers

**Comment by alalazo** [comment]
@tgamblin @scottwittenburg Fyi, this is a case where @spackbot didn't add reviews from maintainer automatically. I have encountered a few in the past, so pointing this out to you in case you have any idea what could be the cause of it.

File: var/spack/repos/builtin/packages/mmg/package.py [hunk 1]
```diff
@@ -58,3 +61,10 @@ def cmake_args(self):
             args.append("-DLIBMMG_STATIC=ON")
 
         return args
+
+    def install(self, spec, prefix):
+        super().install(spec, prefix)
+        # copy source to install path for parmmg
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mmg/package.py]:
```suggestion
    @run_after('install')
    def install_source(self):
        prefix = self.spec.prefix
```

**Review by alalazo — COMMENTED** [review]
I left a minor suggestion. Waiting for maintainer's comments before merging.

**Review by alalazo — DISMISSED** [review]

**Comment by balay** [comment]
> copy source of mmg to installation tree

does parmmg require this [for its build to succeed]? If so - perhaps good to add a comment in source

**Review by balay — DISMISSED** [review]

**Comment by lrtfm** [comment]
> does parmmg require this [for its build to succeed]? If so - perhaps good to add a comment in source

Yes. Done.

**Review by corentin-dev — APPROVED** [review]
Good changes were made, nice work!

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Pull request merged**